packages/trpc/server/routers/viewer/teams/inviteMember/utils.ts

```diff
@@ -136,17 +137,26 @@ export async function createNewUserConnectToOrgIfExists({
   usernameOrEmail,
   input,
   parentId,
+  autoAcceptEmailDomain,
   connectionInfo,
 }: {
   usernameOrEmail: string;
   input: InviteMemberOptions["input"];
   parentId?: number | null;
+  autoAcceptEmailDomain?: string;
   connectionInfo: ReturnType<typeof getOrgConnectionInfo>;
 }) {
   const { orgId, autoAccept } = connectionInfo;
 
+  const [emailUser, emailDomain] = usernameOrEmail.split("@");
+  const username =
+    emailDomain === autoAcceptEmailDomain
+      ? slugify(emailUser)
+      : slugify(`${emailUser}-${emailDomain.split(".")[0]}`);
```

Comment:
This is causing a weird edge case where there might be the possibility that the generated user may exist already. causing a prisma error. Example: we have `robert@example.com` being converted to `robert-example` username. Which already exists in prod. On production this only returns a 500 error:

On dev we get this:

```
Object { message: "\nInvalid `prisma.user.create()` invocation:\n\n\nUnique constraint failed on the fields: (`username,`)", code: -32603, data: {…} }
```